### ISSUE TITLE
Tick speed module

### ISSUE THREAD

**Issue by HSQuillin** [PGMDev/PGM#1211]
`randomTickSpeed` is disabled as a gamerule because it is considered to not be useful. At first glance, this makes sense; it's confusing if a map has different speeds of acceleration of fluids, crops, redstone etc. that are controlled by the game clock. However, I think we can do much better than the vanilla implementation.

I've thought about use cases that make sense in the scope of PGM. What about a minigame where you need to survive fast-decaying leaves or melting snow? Instant trees on a survival-based gamemode by placing saplings so you don't have to wait throughout the normal Minecraft day? Mutations for redstone that make it 2x slower, allowing more time for forward pushes against TNT cannons? 

Some implementation could be as follows using events:

```xml
<ticks>
   <tick speed="50">StructureGrowEvent</tick> <! -- Tick speed of 50 for saplings -->
   <tick speed="0">BlockFadeEvent</tick> <! -- Disable melting of ice in XML w/o taiga biome -->
</ticks>
```

**Concerned?** One of the primary concerns in the past was for fluids,[ this was considered a bug that has since been fixed](https://bugs.mojang.com/browse/MC-93468).

**Comment by Pablete1234** [comment]
If you do not want your ice to melt, disable it via applied regions.

What you're asking here is far, far beyond the scope of random tick speed capabilities (redstone is not goverend by random ticks, neither is tnt) and completely out of scope for PGM, as it requires server-level changes to support different tick speeds for different things affected by tickspeed.

Ontop of that, what you're asking for can often be done with filters and applied regions, you can make a filter for random 10% chance to allow cause=world destroying ice (ie: melting), meaning it becomes, releative to every other random ticked thing, 1/10th as often.

I'm open to allowing randomtickspeed to be changed, and it may actually already be possible (even if docs say otherwise) since we allow arbitrary gamerules to be set (in the past, you could only change a few pre-determined ones). But the whole rest of suggestions are just way too ambitious and out of scope for pgm.

One last thing, you mention a concern about fluids being affected, mentioning it's been fixed. The fix is for 1.12.2. PGM does not even run in 1.12.2 yet.

**Comment by Pablete1234** [comment]
Given the issues i mentioned above and lack of a response, i'm closing the issue

Feel free to re-open or make new issues if you believe there are specific use-cases that are currently not possible with filters
